R/moma_arguments.R

```diff
@@ -389,21 +264,226 @@ create_moma_sparsity_func <- function(f) {
+#' and \code{g}.
+#' @name group_lasso
+#' @references Yuan, Ming, and Yi Lin. "Model Selection and Estimation in Regression
+#' with Grouped Variables." Journal of the Royal Statistical Society:
+#' Series B (Statistical Methodology) 68.1 (2006): 49-67. \doi{10.1111/j.1467-9868.2005.00532.x}.
+#' @inheritParams moma_sparsity_options
+#' @export
 moma_grplasso <- create_moma_sparsity_func(grplasso)
+
+#' Fused LASSO
+#'
+#' Use this function to set the penalty function to fused lasso
+#' \deqn{\lambda \sum | x_{i} - x_{i-1} |,}
+#' where \eqn{\lambda} is set by the \code{lambda} argument below.
+#' @param ... Forces users to specify all arguments by name.
+#' @param algo A string being either "path" or "dp". Defaults to "path". Partial matching
+#' is supported. Two solving algorithms
+#' are provided. When "path" is chosen, the algorithm by
+#' Hoefling, H. (2010) is used. When "dp" is chosen, the algorithm by Johnson, N. A. (2013) is used.
+#'
+#' @references Tibshirani, Robert, et al. "Sparsity and Smoothness via the Fused Lasso."
+#' Journal of the Royal Statistical Society: Series B (Statistical Methodology) 67.1 (2005): 91-108.
+#' \doi{10.1111/j.1467-9868.2005.00490.x}.
+#' @references Hoefling, H. (2010). A path algorithm
+#' for the fused lasso signal approximator. Journal of Computational and Graphical
+#'  Statistics, 19(4), 984-1006. \doi{10.1198/jcgs.2010.09208}.
+#' @references Johnson, N. A. (2013). A dynamic programming algorithm for the
+#' fused lasso and l 0-segmentation. Journal of Computational and Graphical
+#' Statistics, 22(2), 246-260. \doi{10.1080/10618600.2012.681238}.
+#' @return A \code{moma_sparsity_type} object, which is an empty list.
+#'
+#' @name fused_lasso
+#' @inheritParams moma_sparsity_options
+#' @export
 moma_fusedlasso <- create_moma_sparsity_func(fusedlasso)
+
+#' L1 trend filtering
+#'
+#' Use this function to set the penalty function to l1 trend filtering. An
+#' important special case is when \eqn{k=1}. In this case the penalty
+#' term becomes
+#' \deqn{\lambda \sum | x_{i-1} - 2x_{i} + x_{i+1} |,}
+#' where \eqn{\lambda} is set by the \code{lambda} argument below.
+#'
+#' The general formula of the penalty term for \eqn{k \in N} can be found in
+#' the paper cited in Reference. For other values of \eqn{k} please refer to the following table:
+#' \tabular{llll}{
+#'   \tab \eqn{k = 0} \tab \eqn{k = 1} \tab \eqn{k = 2}
+#' \cr
+#'  Type of sparsity \tab piecewise constant \tab peicewise linear \tab piecewise quadratic
+#' }
+#'
+#' @param ... Forces users to specify all arguments by name.
+#' @param l1tf_k Use (\eqn{k+1})-difference matrix in trend filtering. Note \eqn{k = 0}
+#'          implies piecewise constant, \eqn{k=1} implies piecewise linear, \eqn{k=2}
+#'          piecewise quadratic etc.
+#' @references Tibshirani, Ryan J. "Adaptive Piecewise Polynomial Estimation via Trend
+#' Filtering." The Annals of Statistics 42.1 (2014): 285-323. \doi{10.1214/13-AOS1189}.
```

Comment:
You might also cite https://www.tandfonline.com/doi/full/10.1080/10618600.2015.1054033 here as well. (Aren't we using a similar algorithm?)

Reply:
Reference added.

I think so. I was coding the algorithm described one of Ryan Tibshirani's lecture notes. Most likely they are the same thing.